source/extensions/stat_sinks/hystrix/hystrix.h

```diff
@@ -0,0 +1,146 @@
+#pragma once
+
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "envoy/server/admin.h"
+#include "envoy/server/instance.h"
+#include "envoy/stats/stats.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace StatSinks {
+namespace Hystrix {
+
+typedef std::vector<uint64_t> RollingWindow;
+typedef std::map<const std::string, RollingWindow> RollingStatsMap;
+
+struct ClusterStatsCache {
+  // This constructor initializes all the stat names by concatenating
+  // the cluster name and the rest of the stat name that's required for
+  // the lookup.
+  ClusterStatsCache(const std::string& cluster_name);
+
+  std::string cluster_name_;
+
+  // Rolling windows
+  RollingWindow errors_;
+  RollingWindow success_;
+  RollingWindow total_;
+  RollingWindow timeouts_;
+  RollingWindow rejected_;
+
+  void printToStream(std::stringstream& out_str);
+  void printRollingWindow(absl::string_view name, RollingWindow rolling_window,
+                          std::stringstream& out_str);
+};
+
+typedef std::unique_ptr<ClusterStatsCache> ClusterStatsCachePtr;
+
+class HystrixSink : public Stats::Sink, public Logger::Loggable<Logger::Id::hystrix> {
+public:
+  HystrixSink(Server::Instance& server, uint64_t num_of_buckets);
+  HystrixSink(Server::Instance& server);
+  Http::Code handlerHystrixEventStream(absl::string_view, Http::HeaderMap& response_headers,
+                                       Buffer::Instance&, Server::AdminStream& admin_stream);
+  void init();
+  void flush(Stats::Source& source) override;
+  void onHistogramComplete(const Stats::Histogram&, uint64_t) override{};
+
+  /**
+   * register a new connection
```

Comment:
nit: s/register/Registers/ , and specify doxygen for args per convention.

and below
a/remove/Removes/, s/Add/Adds/, etc.